search-enterprise/latest/config/transactional.md

```diff
@@ -0,0 +1,354 @@
+
+When you are upgrading the database, you can add optional indexes in order to support the metadata query feature. This information lets you know the likely duration of the upgrade and how to do it incrementally.
+
+For large repositories, creating the database indexes to support the transactional metadata query can take some time. To check how long it will take, you can add the first index to the database and note the time taken. The full upgrade is estimated to take less than 10 times this value. However, this can vary depending on the structure of the data, the database, and the size of the repository.
+
+The [SQL patch script](http://dev.alfresco.com/resource/AlfrescoOne/5.0/configuration/alfresco/dbscripts/upgrade/4.2/org.hibernate.dialect.Dialect/metadata-query-indexes.sql) can be run in parts, adding one index at a time. The patch is marked complete by the statement that inserts into alf_applied_patch. The patch can be marked as unapplied using the SQL delete statement.
+
+## Configuring search in Alfresco Share
+
+The following sections describe how to configure search in Alfresco Share.
+
+## Controlling permissions checking
+
+You can limit the time Alfresco Content Services spends on ensuring that the user executing the search has the necessary permissions to see each result. Setting this limit increases search speed and reduces the use of resources.
+
+You can limit both the time spent and the number of documents checked before Alfresco Content Services returns a search query using the `system.acl.maxPermissionCheckTimeMillis` and the `system.acl.maxPermissionChecks` properties. The default values are 10000 and 1000 respectively.
+
+1. Open the `<classpathRoot>/alfresco-global.properties` file.
+
+2. Set the `system.acl.maxPermissionCheckTimeMillis` property.
+
+    For example, `system.acl.maxPermissionCheckTimeMillis=20000`.
+
+3. Set the `system.acl.maxPermissionChecks` property.
+
+    For example, `system.acl.maxPermissionChecks=2000`.
+
+    > **Note:** If you increase these values and have a query that returns a very large number of results, (a) the search results will take longer to be returned to the user, and (b) the system will spend longer to check permissions, leading to the possibility of performance degradation. If you set these values to a low number, you run the risk of inconsistent search results every time you run the same search. These settings are also applied when paging. So paging the results will only go up to the maximum returned results based on these settings.
+
+## Controlling search results
+
+Use this information to control the maximum number of items that an Alfresco Share search returns.
+
+By default, the Share search feature returns a maximum of 250 search results. You can extend this number of search results to return more than 250 entries.
+
+1. Download the [share-config.xml](http://dev.alfresco.com/resource/AlfrescoOne/5.1/configuration/alfresco/share-config.xml) file.
```

Comment:
Outdated link.

Reply:
Changed,

Reply:
File is updated.
I have changed 1. to be


1. Download your `share-config.xml` file found at `tomcat/webapps/share/WEB-INF/classes/alfresco/share-config.xml`, for more see [Share Configuration Extension Point](https://docs.alfresco.com/content-services/latest/develop/share-ext-points/share-config/).